src/NuGet.Clients/VisualStudio/Extensibility/CredentialRequestType.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// The type of credential request being made.
+    /// </summary>
+    public enum CredentialRequestType
```

Comment:
In my opinion having a `None` or `Default` value does not make sense here. Credential providers that do no care about the type can simply ignore this parameter. There is not a case that I know of where we prompt for credentials *without* reacting to a 4XX challenge. If we need it later (say, if you want to configure a source as authenticated thus avoiding the 4XX roundtrip), then we can add it.